perun-core/src/main/java/cz/metacentrum/perun/core/entry/MembersManagerEntry.java

```diff
@@ -1578,6 +1578,30 @@ public void updateSponsorshipValidity(PerunSession sess, Member sponsoredMember,
 		membersManagerBl.updateSponsorshipValidity(sess, sponsoredMember, sponsor, newValidity);
 	}
 
+	@Override
+	public List<RichMember> getRichMembersByIds(PerunSession sess, List<Integer> ids, List<String> attrNames) throws PrivilegeException, AttributeNotExistsException {
+		Utils.checkPerunSession(sess);
+
+		// Authorization
+		if (!AuthzResolver.authorizedInternal(sess, "getRichMembersByIds_List<Integer>_List<String>_policy")) {
+			throw new PrivilegeException(sess, "getRichMembersByIds");
+		}
+
+		List<Member> members = getMembersManagerBl().getMembersByIds(sess, ids);
+		members.removeIf(member -> !AuthzResolver.authorizedInternal(sess, "filter-getRichMembersByIds_List<Integer>_List<String>_policy", member));
+
+		List<AttributeDefinition> attributeDefinitions = new ArrayList<>();
+		for (String attrName : attrNames) {
+			attributeDefinitions.add(getPerunBl().getAttributesManagerBl().getAttributeDefinition(sess, attrName));
```

Comment:
I would probably move this before getting members, since if there is non existent attribute definition in the list, we save up DB resources by failing before getting more data.

Reply:
Fixed